app/src/main/java/com/onegini/mobile/exampleapp/network/fcm/FCMListenerService.java

```diff
@@ -41,15 +43,23 @@ public class FCMListenerService extends FirebaseMessagingService {
   public void onMessageReceived(final RemoteMessage message) {
     final OneginiMobileAuthWithPushRequest mobileAuthWithPushRequest = parseOneginiMobileAuthRequest(message);
     if (mobileAuthWithPushRequest != null) {
-      final Intent serviceIntent = getServiceIntent(mobileAuthWithPushRequest);
       if (AppLifecycleListener.isAppInForeground()) {
-        startService(serviceIntent);
+        startService(getIntent(mobileAuthWithPushRequest, MobileAuthenticationService.class));
       } else {
-        NotificationHelper.getInstance(this).showNotification(serviceIntent, mobileAuthWithPushRequest.getMessage());
+        NotificationHelper.getInstance(this)
```

Comment:
Since Android 12 it's not possible to start Service, which then starts an Activity https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines

To adjust, I've changed logic that backgrounded/killed app opens first SplashActivity (in order to start onegini sdk) which passes notification data to LoginActivity, which handles notification.